### PR TITLE
chore: add npm install handler

### PR DESCRIPTION
**Pull request by swatDong**
This is to prepare for customized local debug - to detect whether `npm install` is done and to run `npm install` for each component.
Tests added.

### CODE REVIEW

**Comment by swatDong** [comment]
@1openwindow @HuihuiWu-Microsoft @nliu-ms  You are in reviewer list since there're test changes...
I've adjusted code owner as well but this PR still need your approval.

**Review by HuihuiWu-Microsoft — DISMISSED** [review]

**Comment by HuihuiWu-Microsoft** [comment]
> @1openwindow @HuihuiWu-Microsoft @nliu-ms You are in reviewer list since there're test changes... I've adjusted code owner as well but this PR still need your approval.

Done

**Review by a1exwang — DISMISSED** [review]

**Comment by kuojianlu** [comment]
One question that might be releated to this PR as well as the feature design: since we give back the work of installing packages to user, what if user uses `yarn` instead of `npm`? In this case, there will be no `package-lock.json` but `yarn.lock`.

File: packages/vscode-extension/src/debug/npmInstallHandler.ts
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/npmInstallHandler.ts:10]: `isNpmInstalled` may be more accurate.

Comment (swatDong) [reply packages/vscode-extension/src/debug/npmInstallHandler.ts]: Changed to `checkDependencies` to not confuse with *npm is installed*

**Comment by swatDong** [comment]
> One question that might be releated to this PR as well as the feature design: since we give back the work of installing packages to user, what if user uses `yarn` instead of `npm`? In this case, there will be no `package-lock.json` but `yarn.lock`.

Good catch. Updated to use [@npmcli/arborist](https://www.npmjs.com/package/@npmcli/arborist) to load installed dependencies from `node_modules`, which is the same package used by npm cli.

**Review by kuojianlu — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Pull request merged**
